hashintel/hash
# Implement draft state for entity editor

## 🌟 What is the purpose of this PR?

This PR implements a local draft state stored in the memory for entity editor. So instead of creating tons of useless versions of an entity while editing it, users can manually hit a `save changes` button and save the changes on the entity.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1201095311341924/1203499428268956/f) _(internal)_


## 🐾 Next steps


⚠️ _For now, only properties table is working with draft state. Links table is hidden on `new entity` page_ ⚠️ 

- Implement draft state for links as well
- Implement validation for `create entity` and `save changes` actions


## ❓ How to test this?



1.  Checkout the branch / view the deployment
1. Navigate to "Create entity" page using the dropdown menu on top right of the screen
2. Confirm that the blue bar above works as expected (discard - create actions)
3. Create an entity
4. Confirm that "save changes - discard changes" on the blue bar works as expected

## 📹 Demo


https://user-images.githubusercontent.com/39553853/209305682-c2a2aa39-f6ed-4aff-9936-834717cc4ba2.mov
